adapters/uniqueid_linux.c

```diff
@@ -25,13 +39,38 @@ UNIQUEID_RESULT UniqueId_Generate(char* uid, size_t len)
     }
     else
     {
-        uuid_t uuidVal;
-        uuid_generate(uuidVal);
 
```

Comment:
There is a UUID_generate function in src/uuid.h.  Have you tried using this function instead?  Also, a custom UUID_unparse may be beneficial there.  That way others can use the functionality, and this file has very little change.  Those functions would then need unit tests to accompany them.